Applications/Terminal/Terminal.cpp

```diff
@@ -373,6 +379,42 @@ void Terminal::escape$J(const ParamVector& params)
     }
 }
 
+void Terminal::escape$S(const ParamVector& params)
+{
+    int count = 1;
+    if (params.size() >= 1)
+        count = params[0];
+    dbgprintf("Terminal: Scrolling up %d lines\n", count);
+
+    for (word i = 0; i < count; i++)
+        scroll_up();
+}
+
+void Terminal::escape$T(const ParamVector& params)
+{
+    int count = 1;
+    if (params.size() >= 1)
+        count = params[0];
+    dbgprintf("Terminal: Scrolling down %d lines\n", count);
+
+    for (word i = 0; i < count; i++)
+        scroll_down();
+}
+
+void Terminal::escape$L(const ParamVector& params)
+{
+    int count = 1;
+    if (params.size() >= 1)
+        count = params[0];
+    dbgprintf("Terminal: Adding %d lines below cursor (at line %d)\n", count, m_cursor_row);
+    invalidate_cursor();
+    for (word row = m_rows; row > m_cursor_row; --row)
+        m_lines[row] = m_lines[row - 1];
+    m_lines[m_cursor_row] = new Line(m_columns);
+    ++m_rows_to_scroll_backing_store;
```

Comment:
Shouldn't this be `m_rows_to_scroll_backing_store += count;`?
That variable is for the painting optimization where we know that a chunk of the terminal backing store bitmap can just be shifted vertically upwards to achieve the same result as a normal repaint would.